lib/reversetunnel/transport.go

```diff
@@ -295,6 +322,19 @@ func (p *transport) start() {
 		return
 	}
 
+	var signedHeader []byte
+	goodVersion := dreq.Address == RemoteAuthServer ||
+		utils.CheckVersion(dreq.TeleportVersion, utils.MinIPPropagationVersion) == nil
+	if p.proxySigner != nil && dreq.ClientSrcAddr != nil && dreq.ClientDstAddr != nil && !useTunnel && goodVersion {
```

Comment:
Is there a single place in which we can put this check? It's replicated in `(*localSite).sendSignedPROXYHeader` already.

Reply:
I extracted this check into a function and put near `maybeSendSignedPROXYHeader()`